controllers/gatling_controller_test.go

```diff
@@ -84,5 +86,99 @@ var _ = Context("Inside of a new namespace", func() {
+					Name:      gatlingName,
+					Namespace: ns.Name,
+				},
+				Spec: gatlingv1alpha1.GatlingSpec{
+					GenerateReport:      false,
+					NotifyReport:        false,
+					CleanupAfterJobDone: false,
+					PodSpec: gatlingv1alpha1.PodSpec{
+						Volumes: []corev1.Volume{
+							{
+								Name: "resource-vol",
+								VolumeSource: corev1.VolumeSource{
+									PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+										ClaimName: "resource-pvc",
+									},
+								},
+							},
+						},
+					},
+					PersistentVolumeSpec: gatlingv1alpha1.PersistentVolumeSpec{
+						Name: "resource-pv",
+						Spec: corev1.PersistentVolumeSpec{
+							VolumeMode: &pvFS,
+							AccessModes: []corev1.PersistentVolumeAccessMode{
+								corev1.ReadWriteOnce,
+							},
+							StorageClassName: "",
+							Capacity: corev1.ResourceList{
+								corev1.ResourceStorage: resource.MustParse("1Gi"),
+							},
+							PersistentVolumeSource: corev1.PersistentVolumeSource{
+								Local: &corev1.LocalVolumeSource{Path: "/tmp"},
+							},
+							NodeAffinity: &corev1.VolumeNodeAffinity{
+								Required: &corev1.NodeSelector{
+									NodeSelectorTerms: []corev1.NodeSelectorTerm{
+										{
+											MatchExpressions: []corev1.NodeSelectorRequirement{{Key: "kubernetes.io/os", Operator: corev1.NodeSelectorOpIn, Values: []string{"linux"}}},
+										},
+									},
+								},
+							},
+						},
+					},
+					PersistentVolumeClaimSpec: gatlingv1alpha1.PersistentVolumeClaimSpec{
+						Name: "resource-pvc",
+						Spec: corev1.PersistentVolumeClaimSpec{
+							AccessModes: []corev1.PersistentVolumeAccessMode{
+								corev1.ReadWriteOnce,
+							},
+							VolumeName: "resource-pv",
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{corev1.ResourceStorage: resource.MustParse("1Gi")},
+							},
+						},
+					},
+					TestScenarioSpec: gatlingv1alpha1.TestScenarioSpec{
+						SimulationClass: "PersistentVolumeSampleSimulation",
+						Parallelism:     1,
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "resource-vol",
+								MountPath: "/opt/gatling/user-files/resources/pv",
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, gatling)
+			Expect(err).NotTo(HaveOccurred(), "failed to create test Gatling resource")
+
+			job := &batchv1.Job{}
+			Eventually(func() error {
+				return k8sClient.Get(
+					ctx, client.ObjectKey{Namespace: ns.Name, Name: gatlingName + "-runner"}, job)
+			}).Should(Succeed())
+
+			pv := &corev1.PersistentVolume{}
+			Eventually(func() error {
+				return k8sClient.Get(
+					ctx, client.ObjectKey{Namespace: ns.Name, Name: "resource-pv"}, pv)
+			}).Should(Succeed())
+
+			pvc := &corev1.PersistentVolumeClaim{}
+			Eventually(func() error {
+				return k8sClient.Get(
+					ctx, client.ObjectKey{Namespace: ns.Name, Name: "resource-pvc"}, pvc)
+			}).Should(Succeed())
+		})
+
```

Comment:
Just a question.

Are the below codes needed here?

```
Expect(job.Spec.Parallelism).Should(Equal(pointer.Int32Ptr(1)))
Expect(job.Spec.Completions).Should(Equal(pointer.Int32Ptr(1)))
```

Reply:
Since the code confirms that the results of getGatlingRunnerJobParallelism are correct, we consider it unnecessary in the added test code. Of course, it can be added.